src/main/java/kr/syeyoung/dungeonsguide/dungeon/mechanics/DungeonFairySoul.java

```diff
@@ -39,28 +42,22 @@ public class DungeonFairySoul implements DungeonMechanic {
 
 
     @Override
-    public Set<Action> getAction(String state, DungeonRoom dungeonRoom) {
+    public Set<AbstractAction> getAction(String state, DungeonRoom dungeonRoom) {
         if (!"navigate".equalsIgnoreCase(state)) throw new IllegalArgumentException(state+" is not valid state for secret");
-        Set<Action> base;
-        Set<Action> preRequisites = base = new HashSet<Action>();
-        {
-            ActionInteract actionClick= new ActionInteract(secretPoint);
-            actionClick.setPredicate(PredicateArmorStand.INSTANCE);
-            actionClick.setRadius(3);
-            preRequisites.add(actionClick);
-            preRequisites = actionClick.getPreRequisite();
-        }
-        {
-            ActionMove actionMove = new ActionMove(secretPoint);
-            preRequisites.add(actionMove);
-            preRequisites = actionMove.getPreRequisite();
-        }
-        {
-            for (String str : preRequisite) {
-                if (str.isEmpty()) continue;
-                ActionChangeState actionChangeState = new ActionChangeState(str.split(":")[0], str.split(":")[1]);
-                preRequisites.add(actionChangeState);
-            }
+        Set<AbstractAction> base;
```

Comment:
I liked previous code better. Although it had weird code blocks, you could easily understand what is used where